redfish-core/lib/environment_metrics.hpp

```diff
@@ -0,0 +1,369 @@
+                                 {"SpeedRPM", *attributeValue},
+                                 {"DataSourceUri", tempPath + sensorName},
+                                 {"@odata.id", tempPath + sensorName}});
+                        },
+                        connectionName, validPath,
+                        "org.freedesktop.DBus.Properties", "Get",
+                        "xyz.openbmc_project.Sensor.Value", "Value");
+                }
+                else
+                {
+                    BMCWEB_LOG_DEBUG
+                        << "This is not a fan-related sensor,sensortype = "
+                        << sensorType;
+                    continue;
+                }
+            }
+        },
+        "xyz.openbmc_project.ObjectMapper",
+        "/xyz/openbmc_project/object_mapper",
+        "xyz.openbmc_project.ObjectMapper", "GetSubTree",
+        "/xyz/openbmc_project/sensors", 0, sensorInterfaces);
+}
+
+inline void getPowerWatts(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+                          const std::string& chassisID)
+{
+    const std::array<const char*, 1> totalPowerInterfaces = {
+        "xyz.openbmc_project.Sensor.Value"};
+    const std::string& totalPowerPath =
+        "/xyz/openbmc_project/sensors/power/total_power";
+    crow::connections::systemBus->async_method_call(
+        [asyncResp, chassisID, totalPowerPath](
+            const boost::system::error_code ec,
+            const std::vector<std::pair<std::string, std::vector<std::string>>>&
+                object) {
+            if (ec)
+            {
+                BMCWEB_LOG_DEBUG << "DBUS response error";
+                messages::internalError(asyncResp->res);
+                return;
+            }
+            for (const auto& tempObject : object)
+            {
+                const std::string& connectionName = tempObject.first;
+                crow::connections::systemBus->async_method_call(
+                    [asyncResp, chassisID](const boost::system::error_code ec,
+                                           const std::variant<double>& value) {
+                        if (ec)
+                        {
+                            BMCWEB_LOG_DEBUG << "Can't get Power Watts!";
+                            messages::internalError(asyncResp->res);
+                            return;
+                        }
+
+                        const double* attributeValue =
+                            std::get_if<double>(&value);
+                        if (attributeValue == nullptr)
+                        {
+                            // illegal property
+                            messages::internalError(asyncResp->res);
+                            return;
+                        }
+                        std::string tempPath =
+                            "/redfish/v1/Chassis/" + chassisID + "/Sensors/";
+                        asyncResp->res.jsonValue["PowerWatts"] = {
```

Comment:
PowerWatts: for each loop, do it append value or replace the value?

Reply:
I don't know the behavior, but it has what you need, then I am fine.

Reply:
> PowerWatts: for each loop, do it append value or replace the value?

THB, this loop only run once, because totalPowerInterfaces size is equal to 1.

Reply:
> I don't know the behavior, but it has what you need, then I am fine.

Ack